data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java

```diff
@@ -22,39 +23,65 @@
 import org.jetbrains.annotations.NotNull;
 import tech.pegasys.teku.api.ChainDataProvider;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.response.v1.beacon.GetStateForkResponse;
-import tech.pegasys.teku.api.schema.Fork;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
 import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.infrastructure.json.JsonUtil;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
 import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
+import tech.pegasys.teku.spec.datastructures.state.Fork;
 
-public class GetStateFork extends AbstractHandler implements Handler {
+public class GetStateFork extends MigratingEndpointAdapter {
   private static final String OAPI_ROUTE = "/eth/v1/beacon/states/:state_id/fork";
-  public static final String ROUTE = routeWithBracedParameters(OAPI_ROUTE);
+  public static final String ROUTE = AbstractHandler.routeWithBracedParameters(OAPI_ROUTE);
+
+  private static final SerializableTypeDefinition<Fork> FORK_TYPE =
+      SerializableTypeDefinition.object(Fork.class)
+          .withField("previous_version", BYTES4_TYPE, Fork::getPrevious_version)
+          .withField("current_version", BYTES4_TYPE, Fork::getCurrent_version)
+          .withField("epoch", UINT64_TYPE, Fork::getEpoch)
+          .build();
+
+  private static final SerializableTypeDefinition<StateForkData> RESPONSE_TYPE =
+      SerializableTypeDefinition.object(StateForkData.class)
+          .name("GetStateForkResponse")
+          .description(
+              "The [Fork](https://github.com/ethereum/consensus-specs/blob/v1.0.1/specs/phase0/beacon-chain.md) object from the Eth2.0 spec.")
+          .withField("data", FORK_TYPE, StateForkData::getData)
+          .build();
+
   private final ChainDataProvider chainDataProvider;
 
-  public GetStateFork(final DataProvider dataProvider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
-    this.chainDataProvider = dataProvider.getChainDataProvider();
+  public GetStateFork(final DataProvider dataProvider) {
+    this(dataProvider.getChainDataProvider());
   }
 
-  GetStateFork(final ChainDataProvider chainDataProvider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
+  GetStateFork(final ChainDataProvider chainDataProvider) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getSateFork")
+            .summary("Get state fork")
+            .description("Returns Fork object for state with given 'state_id'.")
+            .tags(TAG_BEACON, TAG_VALIDATOR_REQUIRED)
+            .response(SC_OK, "Request successful", RESPONSE_TYPE)
```

Comment:
404 will be here as 
`.response(SC_NOT_FOUND, "Not found", BadRequest?)` something like that...